nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/conf/logback.xml

```diff
@@ -88,6 +88,17 @@
         </encoder>
     </appender>
 
+    <appender name="DEPRECATION_FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>${org.apache.nifi.bootstrap.config.log.dir}/nifi-deprecation.log</file>
+        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
+            <fileNamePattern>${org.apache.nifi.bootstrap.config.log.dir}/nifi-deprecation_%d.log</fileNamePattern>
+            <maxHistory>30</maxHistory>
+        </rollingPolicy>
+        <encoder class="ch.qos.logback.classic.encoder.PatternLayoutEncoder">
+            <pattern>%date %level [%thread] %logger %msg%n</pattern>
```

Comment:
Do you think the logger name could be shortened in the layout pattern with an upper limit as in other examples?
The abbreviation algorithm with examples can be found [here](https://logback.qos.ch/manual/layouts.html#conversionWord).

Reply:
The logger name is not truncated in this specific scenario because it is important to know the exact class name containing deprecated references.

Under optimal circumstances, the nifi-deprecation.log will be empty, but when a deployment includes deprecated components or features, the logger name and associated stack trace will provide a reference to component usage that needs to be changed.